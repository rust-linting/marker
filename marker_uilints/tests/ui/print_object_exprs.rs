#[derive(Debug, Default)]
pub struct Example {
    a: u32,
}

impl Example {
    fn inc(&mut self, b: u32) {
        self.a += b;
    }

    fn print(&self) {
        println!("{self:#?}");
    }
}

fn main() {
    let mut foo = Example::default();
    let _print_method = foo.print();
    let _print_method = foo.inc(2);
}
