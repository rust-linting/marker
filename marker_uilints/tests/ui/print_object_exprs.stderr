warning: print test
  --> $DIR/print_object_exprs.rs:18:5
   |
18 |     let _print_method = foo.print();
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: Method(
               MethodExpr {
                   data: CommonExprData {
                       _lifetime: PhantomData<&()>,
                       id: ExprId(..),
                       span: SpanId(..),
                   },
                   receiver: Path(
                       PathExpr {
                           data: CommonExprData {
                               _lifetime: PhantomData<&()>,
                               id: ExprId(..),
                               span: SpanId(..),
                           },
                           path: AstQPath {
                               self_ty: None,
                               path_ty: None,
                               path: AstPath {
                                   segments: [
                                       AstPathSegment {
                                           ident: Ident {
                                               name: "foo",
                                               span: $DIR/print_object_exprs.rs:18:25 - 18:28,
                                           },
                                           generics: SynGenericArgs {
                                               args: [],
                                           },
                                       },
                                   ],
                               },
                               target: Var(
                                   VarId(..),
                               ),
                           },
                       },
                   ),
                   method: AstPathSegment {
                       ident: Ident {
                           name: "print",
                           span: $DIR/print_object_exprs.rs:18:29 - 18:34,
                       },
                       generics: SynGenericArgs {
                           args: [],
                       },
                   },
                   args: [],
               },
           )
   = note: `#[warn(marker::test_lint)]` on by default

warning: print test
  --> $DIR/print_object_exprs.rs:19:5
   |
19 |     let _print_method = foo.inc(2);
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: Method(
               MethodExpr {
                   data: CommonExprData {
                       _lifetime: PhantomData<&()>,
                       id: ExprId(..),
                       span: SpanId(..),
                   },
                   receiver: Path(
                       PathExpr {
                           data: CommonExprData {
                               _lifetime: PhantomData<&()>,
                               id: ExprId(..),
                               span: SpanId(..),
                           },
                           path: AstQPath {
                               self_ty: None,
                               path_ty: None,
                               path: AstPath {
                                   segments: [
                                       AstPathSegment {
                                           ident: Ident {
                                               name: "foo",
                                               span: $DIR/print_object_exprs.rs:19:25 - 19:28,
                                           },
                                           generics: SynGenericArgs {
                                               args: [],
                                           },
                                       },
                                   ],
                               },
                               target: Var(
                                   VarId(..),
                               ),
                           },
                       },
                   ),
                   method: AstPathSegment {
                       ident: Ident {
                           name: "inc",
                           span: $DIR/print_object_exprs.rs:19:29 - 19:32,
                       },
                       generics: SynGenericArgs {
                           args: [],
                       },
                   },
                   args: [
                       IntLit(
                           IntLitExpr {
                               data: CommonExprData {
                                   _lifetime: PhantomData<&()>,
                                   id: ExprId(..),
                                   span: SpanId(..),
                               },
                               value: 2,
                               suffix: None,
                           },
                       ),
                   ],
               },
           )

warning: 2 warnings emitted

